### PR TITLE
Refactoring code

### PR DESCRIPTION
**Pull request by iivic**
Separated this change from the initial PR for a cleaner view of the work.

LEARNER-1135

### CODE REVIEW

**Comment by iivic** [comment]
@clintonb please review.

File: ecommerce/coupons/tests/mixins.py [hunk 1]
```diff
@@ -22,6 +22,10 @@ def setUp(self):
         super(CourseCatalogMockMixin, self).setUp()
         cache.clear()
 
+    @staticmethod
+    def build_discovery_catalogs_url(discovery_api_url):
```
Comment (vkaracic) [inline ecommerce/coupons/tests/mixins.py]: I would add a `catalog_id=''` kwarg here since you use that in almost all instances.

**Review by vkaracic — APPROVED** [review]

File: ecommerce/tests/factories.py [hunk 2]
```diff
@@ -35,9 +36,7 @@ class Meta(object):
     send_refund_notifications = False
     enable_sdn_check = False
     enable_embargo_check = False
-    # TODO: Modify discovery_api_url to use Faker().url() and use
-    # mock_access_token_response instead of @mock_course_catalog_api_client decorator
-    discovery_api_url = factory.LazyAttribute(lambda obj: 'http://{domain}/api/v1/'.format(domain=obj.site.domain))
+    discovery_api_url = 'http://{}.fake/'.format(Faker().domain_name())
```
Comment (clintonb) [inline ecommerce/tests/factories.py:40]: Just fake the entire URL.

http://faker.readthedocs.io/en/master/providers/faker.providers.internet.html

**Comment by iivic** [comment]
@clintonb just reminding you to review.

**Review by clintonb — APPROVED** [review]

**Comment by clintonb** [comment]
Remember to squash your commits. Also, check out the guidelines for create an informative commit message: https://openedx.atlassian.net/wiki/display/OpenDev/Git+Best+Practices#GitBestPractices-CommitMessages.

File: ecommerce/tests/factories.py (hunk 2 above)
Comment (iivic) [reply ecommerce/tests/factories.py:40]: I tried the first time and `fake.url()` sometimes creates a url with `https` which will cause the handshake to break in tests. Can't remember the exact issue but I decided that it is less time consuming to use this approach with the same result.

**Pull request merged**